bravado/mapping/spec.py

```diff
@@ -38,17 +56,47 @@ def build(self):
         validator20.validate_spec(self.spec_dict)
         self.api_url = build_api_serving_url(self.spec_dict, self.origin_url)
         self.definitions = build_models(self.spec_dict['definitions'])
+        self.params = self.build_params()
+        self.responses = self.build_responses()
+        self.resources = build_resources(self)
```

Comment:
Same comment about mutable objects.

I think these should all be part of either a separate factory class, or just plain functions on the module which can be used as part of a factory classmethod `from_resource_spec()` or something like that.